lib/mayaUsd/ufe/UsdPointInstanceUndoableCommands.h

```diff
@@ -59,7 +59,10 @@ class MAYAUSD_CORE_PUBLIC UsdPointInstanceUndoableCommandBase : public BaseUfeUn
             return;
         }
 
-        _modifier.setPrimAndInstanceIndex(item->prim(), item->instanceIndex());
+        _modifier.setSceneItem(item);
+        // We're using a modifier to change a point instancer attribute, so
+        // batch the reads and writes, for efficiency.
+        _modifier.joinBatch();
```

Comment:
Why we can't join the batch when setting a scene item on a modifier?

Reply:
That's the way I had it originally, but it's incorrect, as modifiers are used by UsdTransform3dPointInstance to read:
https://github.com/Autodesk/maya-usd/blob/0e13ab44a2f10050ccd257607596539c830980db/lib/mayaUsd/ufe/UsdTransform3dPointInstance.cpp#L49
It's really only when you're using a modifier to write to the attribute that you want to join a batch.  I will add that as a code comment.